lightning-invoice/src/payment.rs

```diff
@@ -367,13 +433,13 @@ where
 				PaymentSendFailure::PathParameterError(_) => Err(e),
 				PaymentSendFailure::AllFailedRetrySafe(_) => {
 					let mut payment_cache = self.payment_cache.lock().unwrap();
-					let retry_count = payment_cache.get_mut(&payment_hash).unwrap();
-					if *retry_count >= self.retry_attempts.0 {
-						Err(e)
-					} else {
-						*retry_count += 1;
+					let payment_attempts = payment_cache.get_mut(&payment_hash).unwrap();
+					if self.retry.is_retryable_now(payment_attempts) {
```

Comment:
Doesn't this break for `Retry::Attempts(0)`? The cache will have a value of `0` which would cause `is_retryable_now` to return true. But in this case we wouldn't want to the retry since `InvoicePayer` is configured for no retries. Perhaps we need a test for this case?

Reply:
Sure, let's just move the increment up. Upstream, the check has the operands flipped with the opposite meaning (error), so works as intended.

Reply:
ah good catch. I have moved the increment up.